Tests/MaterialLib/TestMPLParseMaterial.cpp

```diff
@@ -0,0 +1,322 @@
+/// material or of a specifier and to store them in two vectors for later
+/// comparison.
+void getNames(MPL::Property const& observation, std::string expectation,
+              std::string defaultName, std::vector<std::string>* obs,
+              std::vector<std::string>* exp)
+{
+    obs->push_back(MPL::getString(observation));
+    if (expectation == "")
+        exp->push_back(defaultName);
+    else
+        exp->push_back(expectation);
+}
+
+// This test represents an invariant test. First, several phases,
+// components, and properties are generated (more or less randomly).
+// Then, an XML-tree is generated from those information. The XML
+// tree is then parsed into a configTree, from which an
+// MPL::MaterialObject is generated.
+// Last step compares the names (as well as the topology) of the
+// Material object with the specified parameters.
+TEST(Material, parseMaterials)
+{
+    // This is the topology of our new material: The size of the
+    // topology vector determines the number of phases, while each
+    // vector component refers to the number of components of that
+    // phase.
+    // The number of properties is fixed in each case and is
+    // determined by the size of the PropertyEnum enumerator.
+    std::vector<std::size_t> const mediumTopology = {1, 3, 2};
+
+    // make a string from every property enumerator
+    std::vector<std::string> property = MPL::convertEnumToString;
+
+    Medium medium(mediumTopology);
+
+    // the omnivagant medium:
+    medium.property[MPL::name] = "luminiferous_aether";
+
+    medium.phases[0].property[MPL::name] = "solid";
+    medium.phases[1].property[MPL::name] = "liquid";
+    medium.phases[2].property[MPL::name] = "gas";
+
+    medium.phases[0].component[0].property[MPL::density] = "LinearTemperature";
+    medium.phases[0].component[0].property[MPL::thermal_conductivity] = "0.654";
+    medium.phases[0].component[0].property[MPL::reference_temperature] = "333";
+    medium.phases[0].component[0].property[MPL::reference_density] = "2100.0";
+    medium.phases[0].component[0].property[MPL::drhodT] = "-0.4";
+    medium.phases[0].component[0].property[MPL::effective_stress] =
+        "LinearEpsilon";
+
+    medium.phases[1].component[0].property[MPL::name] = "Water";
+    medium.phases[1].component[1].property[MPL::name] = "CarbonDioxide";
+    medium.phases[1].component[2].property[MPL::name] = "SodiumChloride";
+
+    medium.phases[1].property[MPL::density] = "Duan_2012";
+    medium.phases[1].property[MPL::viscosity] = "Islam_Carlson_2012";
+
+    medium.phases[2].component[0].property[MPL::name] = "Water";
+    medium.phases[2].component[0].property[MPL::viscosity] = "IAPWS_2008";
+    medium.phases[2].component[1].property[MPL::name] = "CarbonDioxide";
+    medium.phases[2].component[1].property[MPL::viscosity] = "Fenghour_1998";
+
+    medium.phases[2].property[MPL::density] = "Peng_Robinson_1976";
```

Comment:
Need some tests to test getValue() of the complex models in this PR.  For example, the documentation of IAWPS has a test example, which I have implemented before to the source code (you can copy it from https://github.com/ufz/ogs/blob/master/Tests/MaterialLib/TestFluidViscosity.cpp#L130).